### PR TITLE
OADP-2049: Cleanup replication source CRs if retain policy not set

### PR DESCRIPTION
**Pull request by shubham-pampattiwar**

### CODE REVIEW

File: controllers/cleanup.go [hunk 1]
```diff
@@ -63,6 +63,17 @@ func (r *VolumeSnapshotBackupReconciler) CleanBackupResources(log logr.Logger) (
...
 
+	// Check if retain policy is set by the user for datamover backups
+	retainPolicyPresent, err := r.isRetainPolicySet(&vsb)
+	if err != nil {
+		return false, err
+	}
+
+	// If retainPolicy is not set then cleanup the ReplicationSource objects as well
+	if !retainPolicyPresent {
+		cleanupVSBTypes = append(cleanupVSBTypes, &volsyncv1alpha1.ReplicationSource{})
```
Comment (sseago) [inline controllers/cleanup.go:74]: This var is defined outside of the function, so it's not tied to a particular run. Does that mean we add ReplicationSource when we run CleanBackupResources right now (if retain policy is set), but then later, we run it again without retain policy, even though we're not adding it again, it's already there and therefore we're still acting as if the retain policy was in place. Do we need to move the `var cleanupVSBTypes` declaration into the func?

**Comment by weshayutin** [comment]
@shubham-pampattiwar thank you! leaving the code review to the experts

File: controllers/cleanup.go (hunk 1 above)
Comment (shubham-pampattiwar) [reply controllers/cleanup.go:74]: yeah you are right, this is basically per DPA config, so if the config changes then the `var cleanupVSBTypes` should also change with it.

Comment (shubham-pampattiwar) [reply controllers/cleanup.go:74]: Done, updated the PR

**Review by sseago — APPROVED** [review]

**Pull request merged**